shared/cert.go

```diff
@@ -20,9 +21,126 @@ import (
 	"os"
 	"os/user"
 	"path"
+	"path/filepath"
 	"time"
 )
 
+// KeyPairAndCA returns a CertInfo object with a reference to the key pair and
+// (optionally) CA certificate located in the given directory and having the
+// given name prefix
+//
+// The naming conversion for the various files is:
+//
+// <prefix>.crt -> public key
+// <prefix>.key -> private key
+// <prefix>.ca -> CA certificate
+//
+// If no public/private key files are found, a new key pair will be generated
+// and saved on disk.
+//
+// If a CA certificate is found, it will be returned as well as second return
+// value (otherwise it will be nil).
+func KeyPairAndCA(dir, prefix string, kind CertKind) (*CertInfo, error) {
+	certFilename := filepath.Join(dir, prefix+".crt")
+	keyFilename := filepath.Join(dir, prefix+".key")
+
+	// Ensure that a server certificate exists, or create a new one if it
```

Comment:
server? This can generate either.

Reply:
Good catch, fixed in b917f33703fa53871935e06ae84465d2da48b430.